File: FrontEnd/src/services/UserService.js
import axios from 'axios';

const USER_API_BASE_URL = "http://localhost:8080/api/v1/user";

class UserService {
  getUsers() {
    return axios.get(`${USER_API_BASE_URL}/getusers`);
  }

  addUser(user) {
    return axios.post(`${USER_API_BASE_URL}/adduser`, user);
  }

  updateUser(user) {
    return axios.put(`${USER_API_BASE_URL}/updateuser`, user);
  }

  deleteUser(user) {
    return axios.delete(`${USER_API_BASE_URL}/deleteuser`, { data: user });
  }
}

const userServiceInstance = new UserService();
export default userServiceInstance;

File: BackEnd/src/main/java/com/tdStore/TD/Store1/service/UserService.java
package com.tdStore.TD.Store1.service;

import com.tdStore.TD.Store1.dto.UserDto;
import com.tdStore.TD.Store1.model.User;
import com.tdStore.TD.Store1.repo.UserRepo;
import org.modelmapper.ModelMapper;
import org.modelmapper.TypeToken;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@Transactional
public class UserService {
    @Autowired
    private UserRepo userRepo;

    @Autowired
    private ModelMapper modelMapper;

    public List<UserDto> getAllUsers(){
        List<User> userList = userRepo.findAll();
        return modelMapper.map(userList, new TypeToken<List<UserDto>>(){}.getType());
    }

    public UserDto addUser(UserDto userDto){
        userRepo.save(modelMapper.map(userDto, User.class));
        return userDto;
    }

    public UserDto updateUser(UserDto userDto){
        userRepo.save(modelMapper.map(userDto, User.class));
        return userDto;
    }

    public String deleteUser(UserDto userDto){
        userRepo.delete(modelMapper.map(userDto, User.class));
        return "User Deleted";
    }
}
